File: eCinema/eCinema.Model/Entities/Booking.cs
using CinemaApp.Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading.Tasks;

namespace eCinema.Models.Entities
{
    public class Booking
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int ShowtimeId { get; set; }
        public DateTime BookingTime { get; set; }
        public string? DiscountCode { get; set; }
        public User User { get; set; }
        public Showtime Showtime { get; set; }
        public Payment Payment { get; set; }
        public ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
        public ICollection<BookingConcession> BookingConcessions { get; set; } = new List<BookingConcession>();
    }
}

File: eCinema/eCinema.UI/ecinema_mobile/lib/models/booking.dart
import 'package:json_annotation/json_annotation.dart';
import 'ticket.dart';
import 'booking_concession.dart';

part 'booking.g.dart';

@JsonSerializable(explicitToJson: true)
class Booking {
  final int id;
  final int userId;
  final int showtimeId;
  final DateTime bookingTime;
  final String? discountCode;
  final List<Ticket> tickets;
  final List<BookingConcession> bookingConcessions;

  Booking({
    required this.id,
    required this.userId,
    required this.showtimeId,
    required this.bookingTime,
    this.discountCode,
    required this.tickets,
    required this.bookingConcessions,
  });

  factory Booking.fromJson(Map<String, dynamic> json) =>
      _$BookingFromJson(json);

  Map<String, dynamic> toJson() => _$BookingToJson(this);
}
